### DISCUSSION TITLE
Some ideas for the trip

### DISCUSSION THREAD

**Discussion by umpfmpf** [devemux86/cruiser#581] [Q&A] [answered]
I've been driving through Wales and southern France for a few weeks now. 
I missed the following things: 
- When clicking on the route symbol at the top left, it would practically be a button to switch between the current section profile and a set replacement profile, for example the fastest route by simply pressing a button (when I get closer to a larger town, I don't necessarily want to drive extra curvy, but drive through as quickly as possible. After passing through, continue curving again) 
- Importing a list of blocked passes in order to import them into the navigation as closed roads is easy to create, for example on alpenpaesse.de/verkehrsinfo). 
- [poi_example_alpenpaesse.csv](https://github.com/user-attachments/files/16043739/poi_example_alpenpaesse.csv)
- When navigating offline with a brouter and setting curvy routes, a small detour is often installed in small parking spaces or in town thoroughfares, which I can only eliminate with various settings when I change the route planning towards a fast and curvy route. 

- Otherwise I have to say that the driving is very clear with Cruiser.

**Comment by devemux86** [comment]
@umpfmpf thanks for your report!

- Do you mean the top left direction icon in navigation that shows actions or options (via press or long press)?
You want to change the current section profile to be used in the rerouting during navigation?

- Import blocked passes so they are blocked in routing or just to see them on the map?

- BRouter: can you post a small route example and your profile settings, so we can check the route?

**Reply by umpfmpf** [reply]
@demux86 

- Yes, via the long press Menü. And yes, i want to change the Profile only for this section and then change it back to the Profile used before. I'd need only a quick Option to pass the next City, Area, to use the fastest way to my Endpoint instead of the slower curvy roads that i normally use or to drive for a while on larger roads if it starts to rain. 

- To use the blocked Passes in routing and planing. This means I can load and import a new file in the morning before the trip without experiencing any unpleasant surprises on the trip.

- 
[Motorrad.json](https://github.com/user-attachments/files/16044086/Motorrad.json)
[Beispiel.json](https://github.com/user-attachments/files/16044087/Beispiel.json)

**Reply by WolfgangZ42** [reply]
Yesterday I was on road (by 32°C - puuuh) - and we had some detours as well. Didn't record the tour, but it took few minutes to plan it new. I marked the particular detours by 7 shaping points. Imported the resulting gpx file into Kurviger (w/o any change) and exported it again. Here are the files:  [Blautopf.zip](https://github.com/user-attachments/files/16044114/Blautopf.zip)
Btw: It's not dramatically - I look after the tour in planning anyway and set shaping points - but yesterday I forgot.

**Reply by devemux86** [reply]
@umpfmpf 

- If you mean just for the rerouting in navigation,
we could add a rerouting option to use the route profile or fastest route.

- Any content, like blocked passes, can be imported using the popular GPX format.
You can also store them in favorites, so you can have them always available.

- In this example, do you mean to skip the Henfynyw village?
If we move the start / end away from the village and select "curvy" route or avoid motorways + trunk roads,
then the village is bypassed:

![Route](https://github.com/devemux86/cruiser/assets/3484020/41dabfd1-5dfb-43bb-be10-7e211adfa6dc)

**Reply by umpfmpf** [reply]
- The fact that rerouting simply switches over and when I think that it is enough with fast now, then I can go back to the route profile myself.
- If I import the blocked passes as gpx and then insert them into Brouter as blocked areas, it still leads over the blocked pass, even though it is set in the options that it does not ignore this.
- I mean this small redirection. And i have this kind of small redirections realy often. Sometimes through a parking lot.
![Umleitung2](https://github.com/devemux86/cruiser/assets/105640513/0ef781b8-dd6c-4b60-9f93-12b18dfa25af)

**Comment by devemux86** [comment, marked as answer]
- No-go areas in route planning were introduced in https://github.com/devemux86/cruiser/discussions/641

- Route example: it depends on the waypoints position and the routing options.
If we place the previous / next waypoints near the village, then routing cannot bypass it, it just skips some parts of motorway.
Try to move the previous / next waypoints away from the cities / villages so the routing can bypass them (like the above [image](https://github.com/devemux86/cruiser/discussions/581#discussioncomment-9916479)).

**Reply by devemux86** [reply]
> I mean this small redirection. And i have this kind of small redirections realy often. Sometimes through a parking lot.

This redirection can happen when using "Avoid trunk roads" and the previous / next waypoints are on this trunk road.

**Reply by devemux86** [reply]
It is about this trunk road: https://www.openstreetmap.org/way/346886442

**Reply by umpfmpf** [reply]
Okay. If I change the "Avoid trunk roads"strength to 2, this redirection no longer happens.

**Reply by MotoUKRider** [reply]
I wonder if a "minimum-diversion" value could be hard-coded into the routine to prevent *silly* detours like this?  Perhaps 1mi/1km?  It seems kind of obvious (to a human) that riding off a Trunk and then back onto it 100yrds later is less than ideal.

**Reply by devemux86** [reply]
@MotoUKRider do you mean in the BRouter profiles?
(if they support such function)

**Reply by MotoUKRider** [reply]
@devemux86 Yes, something to prevent it from making such a short alteration just for the sake of adhering to the balance/weighting of the Avoidance settings.

**Reply by devemux86** [reply]
It's important to remember that machines do not know what we think and like. They just follow our orders!
In this case, if we set the waypoints on a trunk road and use avoid trunk roads, only deviations are allowed.
Otherwise the routing will be impossible.

**Reply by 0709wiwiwi** [reply]
Short nonsensical deviations ? It's something I've also noticed recently through more routers.
I'm just reporting this. Reporting such items regularly is all I can do.
[Strange routing results](https://discuss.graphhopper.com/t/strange-routing-result/8736)

**Comment by umpfmpf** [comment]
I appreciate the advantages of Cruiser, that a lot of things can be adapted and adjusted. 

But when it comes to navigation, you can get lost. 
- If I change the speed in Brouter something happens. 
- Something happens when changing the avoidance of different types of roads. 
- Something happens when avoiding or favoring rivers, forests or cities. 

But the interaction of all these components has now become so difficult that newcomers in particular are directly deterred and older users also have to struggle with it. 

Maybe it would be a good step in this jungle of setting options to provide a little more overview (e.g. the speed in the Brouter).

**Reply by devemux86** [reply]
@umpfmpf 

What you describe is about calculation of routes (not navigation).

> Something happens when changing the avoidance of different types of roads.

- Roads have different types.
Routing services provide options to avoid certain road types, e.g. motorways, primary roads.

About the other options, they have descriptions, where exactly needs more explanation?

- BRouter speed: it tries to avoid high speed roads.
- Cities, forests, ...: avoid / prefer specific area types.

---

It is better to use the main avoid options and not the options in the "More" screen.
(they are separate because they are more advanced or experimental)

Except avoid cities, I read many complaints about the other "More" options.
If they are confusing, we can remove some to simplify them. :slightly_smiling_face:
